### DISCUSSION TITLE
iMessage database not found in iPhone backup

### DISCUSSION THREAD

**Discussion by michenator19** [ReagentX/imessage-exporter#324] [Q&A] [answered]
Starting a new discussion from my question on another thread. Seems that in my standard iOS backup, `imessage-exporter` is not finding the iMessages database in the expected place. I'm seeing others have this problem in various other contexts, but the answer was that the backup was encrypted. In my case the backup is not encrypted.

Here is the command I put in:

`imessage-exporter -f html -p ~/Desktop/iphone_backup_latest/00008101-001A15682644001E -a iOS -c compatible -o ~/Desktop/iMessages1`

Here is the response I get back:

`Invalid configuration: Database not found at /Users/michele/Desktop/iphone_backup_latest/00008101-001A15682644001E/3d/3d0d7e5fb2ce288813306e4d4636395e047a3d28`

iPhone 12 mini is running iOS 17.5.1 and MacBook Air is running Sonoma 14.6.1. I used the step-by-step from [this Reddit comment](https://www.reddit.com/r/macapps/comments/14jy8yf/comment/jpp4f21/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button) to install (subbing out ReagentX instead of aaronkelly, which did not work) and the step-by-step from [this Github comment](https://github.com/ReagentX/imessage-exporter/discussions/171#discussioncomment-7075177) to understand how to tweak for iOS backup. Happy to share any other details! Thank you!

**Comment by ReagentX** [comment]
I wouldn't follow the steps in that Reddit post; it looks AI generated and incorrect. All you need to to do install it is follow the steps [here](https://github.com/ReagentX/imessage-exporter/tree/develop/imessage-exporter#installation). 

https://github.com/ReagentX/imessage-exporter/blob/511d17d217ab4d7708632ec0a706db09eeb4f12d/imessage-database/src/tables/table.rs#L76-L80

To better diagnose, can you show me the output when you run `ls -la ~/Desktop/iphone_backup_latest`? Also, can you try running `imessage-exporter` without the `-a` flag? It should automatically detect if the database is from an iOS or macOS device, but the output may help us understand what is going wrong.

**Reply by michenator19** [reply, marked as answer]
Deeply embarrassing and so sorry for the time waste but I figured out the issue was a silly typo in my folder name. Closed this discussion, but feel free to delete! Thank you for this program, it's incredible!
